crates/astria-composer/src/collectors/grpc.rs

```diff
@@ -0,0 +1,139 @@
+// ! `GrpcCollector` allows users to directly send Rollup transactions to the Composer
+//
+// ! The [`GrpcCollector`] listens for incoming gRPC requests and sends the Rollup
+// transactions to the ! Executor. The Executor then sends the transactions to the Astria
+// Shared Sequencer.
+
+use std::{
+    net::SocketAddr,
+    time::Duration,
+};
+
+use astria_core::{
+    generated::composer::v1alpha1::{
+        grpc_collector_service_server::{
+            GrpcCollectorService,
+            GrpcCollectorServiceServer,
+        },
+        SubmitRollupTxsRequest,
+    },
+    sequencer::v1::{
+        asset::default_native_asset_id,
+        transaction::action::SequenceAction,
+        RollupId,
+    },
+};
+use astria_eyre::{
+    eyre,
+    eyre::WrapErr,
+};
+use tokio::{
+    io,
+    net::TcpListener,
+    sync::mpsc::error::SendTimeoutError,
+};
+use tonic::{
+    Request,
+    Response,
+};
+
+use crate::{
+    executor,
+    executor::Handle,
+};
+
+/// `GrpcCollector` listens for incoming gRPC requests and sends the Rollup transactions to the
+/// Executor. The Executor then sends the transactions to the Astria Shared Sequencer.
+///
+/// It implements the `ComposerService` rpc service.
+///
+/// The composer will only have one `GrpcCollector` running at a time.
+pub(crate) struct Grpc {
+    listener: TcpListener,
+    executor_handle: executor::Handle,
+}
+
+impl Grpc {
+    pub(crate) async fn new(
+        grpc_addr: SocketAddr,
+        executor_handle: executor::Handle,
+    ) -> eyre::Result<Self> {
+        let listener = TcpListener::bind(grpc_addr)
+            .await
+            .wrap_err("failed to bind grpc listener")?;
+
+        Ok(Self {
+            listener,
+            executor_handle,
+        })
+    }
+
+    /// Returns the socket address the grpc collector is served over
+    /// # Errors
+    /// Returns an error if the listener is not bound
+    pub(crate) fn local_addr(&self) -> io::Result<SocketAddr> {
+        self.listener.local_addr()
+    }
+
+    pub(crate) async fn run_until_stopped(self) -> eyre::Result<()> {
+        let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
+
+        let composer_service = GrpcCollectorServiceServer::new(self.executor_handle);
+        let grpc_server = tonic::transport::Server::builder()
```

Comment:
With the health service added (and this changing the nature of composer so much), we should have a root-level `grpc` module which is the actual gRPC server. `collectors::grpc` can then just contain the implementation of `GrpcCollectorServiceServer`.

So basically I think this logic should be moved there.